c2corg_ui/static/js/feed.js

```diff
@@ -0,0 +1,153 @@
+goog.provide('app.FeedController');
+goog.provide('app.feedDirective');
+
+goog.require('app');
+goog.require('app.Api');
+goog.require('app.Authentication');
+goog.require('app.utils');
+
+
+/**
+ * @return {angular.Directive} The directive specs.
+ */
+app.feedDirective = function() {
+  return {
+    restrict: 'A',
+    controller: 'appFeedController as feedCtrl'
+  };
+};
+app.module.directive('appFeed', app.feedDirective);
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @param {app.Authentication} appAuthentication
+ * @param {app.Api} appApi Api service.
+ * @param {angular.$sce} $sce Angular Strict Contextual Escaping
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.FeedController = function($scope, $attrs, appAuthentication, appApi, $sce) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+  /**
+   * @type {app.Authentication}
+   * @private
+   */
+  this.auth_ = appAuthentication;
+
+  /**
+   * @type {angular.$sce} $sce Angular Strict Contextual Escaping
+   * @private
+   */
+  this.sce_ = $sce;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.location_ = $attrs['location'];
+
+  /**
+   * @type {Array<Object>}
+   * @export
+   */
+  this.documents = [];
+
+  /**
+   * @type {string | undefined}
+   * @private
+   */
+  this.nextToken_;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.busy = true;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.feedType_;
+
+  switch (this.location_) {
+    case 'home':
+      this.feedType_ = this.auth_.isAuthenticated() ? 'personal' : 'standard';
+      break;
+    case 'user-profile':
+      this.feedType_ = 'profile';
+      break;
+    default:
+      break;
+  }
+
+  this.getDocumentsFromFeed();
+};
+
+
+/**
+ * Fills the feed with documents.
+ * Used by ng-infinite-scroll directive in the template.
+ * @export
+ */
+app.FeedController.prototype.getDocumentsFromFeed = function() {
+  this.busy = true;
+  this.api_.readFeed(this.feedType_, this.nextToken_).then(function(response) {
+    var data = response['data']['feed'];
+    for (var i = 0; i < data.length; i++) {
+      this.documents.push(data[i]);
+    }
+    this.nextToken_ = response['data']['pagination_token'];
```

Comment:
From the idea yes, but I would prefer a proper if-else construct instead of doing side-effects in the ternary operator:

```
if (token) {
   this.nextToken_ = token;
   this.busy = false;
} else {
   this.busy = true; // disable scroll
}